bridge_adaptivity/module/views.py

```diff
@@ -269,10 +292,77 @@ class CollectionUpdate(BaseCollectionView, SetUserInFormMixin, ModalFormMixin, U
     pass
 
 
+@method_decorator(login_required, name='dispatch')
+class CollectionOrderUpdate(
+    BaseCollectionOrderView,
+    SetUserInFormMixin,
+    CollectionOrderEditFormMixin,
+    ModalFormMixin,
+    UpdateView,
+):
+
+    def get_object(self):
+        collection_order = CollectionOrder.objects.get(slug=self.kwargs.get("collection_order_slug"))
+        self.collection_order_group_slug = collection_order.group.slug
+        return collection_order
+
+    def get_success_url(self):
+        return reverse("module:group-detail", kwargs={'group_slug': self.collection_order_group_slug})
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['user'] = self.request.user
+        kwargs['group'] = self.object.group
+        kwargs['read_only'] = self._set_read_only_collection()
+        return kwargs
+
+    def _set_read_only_collection(self):
+        return bool(self.kwargs.get('collection_order_slug'))
+
+
+class CollectionOrderAdd(
+    BaseCollectionOrderView,
+    SetUserInFormMixin,
+    CollectionOrderEditFormMixin,
+    ModalFormMixin,
+    CreateView
+):
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['user'] = self.request.user
+        kwargs['group'] = get_object_or_404(ModuleGroup, slug=self.kwargs.get('group_slug'))
+        kwargs['read_only'] = self._set_read_only_collection()
+        return kwargs
+
+    def get(self, request, *args, **kwargs):
+        """
+        Handle GET requests: instantiate a blank version of the form.
+        """
+        result = super().get(request, *args, **kwargs)
+        result.context_data["group"] = get_object_or_404(ModuleGroup, slug=self.kwargs.get('group_slug'))
+        result.context_data['form'].fields['collection'].required = False
+        result.context_data['collection_form'].fields['owner'].initial = self.request.user.id
+        return result
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.request.POST.get('collection_group-collection'):
+            context['collection_form'] = None
+        return context
+
+    def get_success_url(self):
+        return reverse("module:group-detail", kwargs={'group_slug': self.kwargs.get("group_slug")})
+
+    def _set_read_only_collection(self):
```

Comment:
I would use static method here

Reply:
This method overrides realization 'get_success_url' method in  'django.views.generic.edit.py:FormMixin'
Chain of inheritance:
CollectionOrderAdd -> CreateView -> BaseCreateView -> ModelFormMixin -> FormMixin